code/src/main/kotlin/com/expediagroup/sdk/v2/core/apache/util/ApacheHttpTransportUtil.kt

```diff
@@ -1,35 +1,66 @@
 package com.expediagroup.sdk.v2.core.apache.util
 
-import com.expediagroup.sdk.v2.core.trait.common.IdTrait
 import com.expediagroup.sdk.v2.core.trait.configuration.ClientConfiguration
 import com.google.api.client.http.apache.v2.ApacheHttpTransport
-import java.util.*
-import java.util.concurrent.ConcurrentHashMap
-import java.util.concurrent.ConcurrentMap
 
+/**
+ * Creates an `ApacheHttpTransport` instance based on the provided `ClientConfiguration`.
+ *
+ * @param configuration The `ClientConfiguration` used to configure the `ApacheHttpTransport` instance.
+ * @return A new instance of `ApacheHttpTransport` configured according to the provided `ClientConfiguration`.
+ */
 fun createApacheHttpTransport(configuration: ClientConfiguration): ApacheHttpTransport =
-    ApacheHttpTransport(getSingletonHttpClient(configuration))
+    ApacheHttpTransport(createHttpClient(configuration))
 
+/**
+ * Retrieves a singleton instance of `ApacheHttpTransport` based on the provided `ClientConfiguration`.
+ *
+ * @param configuration The `ClientConfiguration` used to configure the `ApacheHttpTransport` instance.
+ * @return An instance of `ApacheHttpTransport`.
+ */
 fun getSingletonApacheHttpTransport(configuration: ClientConfiguration) =
     CreateSingletonApacheHttpTransportLambda.execute(configuration)
 
+/**
+ * A lambda class responsible for creating a singleton instance of `ApacheHttpTransport`.
+ * This class implements the function type `(ClientConfiguration) -> ApacheHttpTransport`,
+ * and provides thread-safe creation of a singleton `ApacheHttpTransport` instance based on
```

Comment:
How is this code thread safe? I don't see any synchronization on the singleton retrieval.
I did some further research on this, and my conclusion is - if we want to insure thread-safety - we can refactor the code here as follows:
```kotlin
fun createApacheHttpTransport(configuration: ClientConfiguration): ApacheHttpTransport =
    ApacheHttpTransport(createHttpClient(configuration))

fun getSingletonApacheHttpTransport(configuration: ClientConfiguration): ApacheHttpTransport =
    ApacheHttpTransportSingleton.getInstance(configuration)

private object ApacheHttpTransportSingleton {
    @Volatile
    private var transport: ApacheHttpTransport? = null

    fun getInstance(configuration: ClientConfiguration): ApacheHttpTransport {
        return transport ?: synchronized(this) {
            transport ?: createApacheHttpTransport(configuration).also { transport = it }
        }
    }
}
```
In the code snippet above, we are adding a synchronized block on retrieving the singleton. Also, marking the transport property with `@Volatile` to prevent thread local caching of the transport property.

> The @Volatile annotation in Kotlin is used to mark a variable as volatile, which means that reads and writes to that variable are immediately visible to all threads. In other words, it prevents threads from caching the variable locally, ensuring that any changes made to it are propagated predictably across all threads.

Another thing to note is the the "Double-Checked Locking", where we are checking for the `transport` nullability twice, outside the synchrinzed and again inside it  to reduce synchronization overhead. The first check `transport ?:` is performed without locking for performance. The second check inside the synchronized block ensures that only one thread initializes transport.

Reply:
Done.